addons/mail/static/src/core/common/composer.js

```diff
@@ -639,4 +664,24 @@ export class Composer extends Component {
             this.threadService.markAsRead(this.props.composer.thread);
         }
     }
+
+    saveInputToLocalStorage() {
+        const fullComposerContent =
+            document
+                .querySelector(".o_mail_composer_form_view #body_0")
+                ?.innerText.replace(/(\t|\n)+/g, "\n") ?? this.props.composer.textInputContent;
+        browser.localStorage.setItem(
+            `${this.thread.localId}_fullComposerText`,
+            fullComposerContent
+        );
+    }
+
+    setInputFromLocalStorage() {
+        const fullComposerContent = browser.localStorage.getItem(
+            `${this.thread.localId}_fullComposerText`
+        );
+        if (fullComposerContent) {
+            this.props.composer.textInputContent = fullComposerContent;
+        }
+    }
```

Comment:
(minor) I would rename as follow:
- `saveInputToLocalStorage` => `saveContent`
- `setInputFromLocalStorage` => `restoreContent`

_Reasoning: simpler name make code easier to read_